ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1397,7 +1399,24 @@ private void fetchRequests(RequestRegistrationCallback<PutOperation> requestRegi
     protected PutRequest createPutRequest() {
       return new PutRequest(NonBlockingRouter.correlationIdGenerator.incrementAndGet(), routerConfig.routerHostname,
           chunkBlobId, chunkBlobProperties, ByteBuffer.wrap(chunkUserMetadata), buf.retainedDuplicate(),
-          buf.readableBytes(), BlobType.DataBlob, encryptedPerBlobKey != null ? encryptedPerBlobKey.duplicate() : null);
+          buf.readableBytes(), BlobType.DataBlob, encryptedPerBlobKey != null ? encryptedPerBlobKey.duplicate() : null,
+          getCrc32Implementation());
+    }
+
+    /**
+     * Select crc32 implementation based on {@link RouterConfig#routerPutRequestCrc32Implementation}.
+     * @return
+     */
+    protected PutRequestCrc32Impl getCrc32Implementation() {
```

Comment:
Can we make it easy: just test Ture or False?
```
if(routerConfig.useJaveNativeCrc32) {
    return PutRequestCrc32Impl.getJavaNativeInstance();
} else {
    return PutRequestCrc32Impl.getAmbryUtilInstance();
}
```

I don't think we will introduce any other crc32 impl in the future, do we?

Reply:
updated